src/pages/resources/Resources.js

```diff
@@ -1,24 +1,33 @@
-import ExternalLink from "../../components/externalLink/ExternalLink"
+import React from "react";
+import "./Resources.css";
+import { resources } from "../../utils/constant";
 
-function Resources() {
+const Resources = () => {
   return (
-    <div>
-      <h1>Useful Resources</h1>
-      <div>The following links might be of interest to club members</div>
-      <br />
-      <div className="links">
-        <ExternalLink 
-          url='https://gdsc.community.dev/conestoga-college-waterloo-campus/' 
-          explanation='GDSC Portal - a place where all of our events are hosted' />
-        <ExternalLink 
-          url='https://conestogastudents.com/2021/09/01/coding-club/' 
-          explanation='CSI Club - register for the official GDSC Conestoga CSI club' />
-        <ExternalLink 
-          url='https://www.youtube.com/channel/UCBbpXl_fLG6dbu25nu2nRNA' 
-          explanation='YouTube - our channel with event recordings' />
+    <div className="resources_wrapper">
+      <div className="top">
+        <h1>OUR RESOURCES</h1>
+      </div>
+      <div className="bottom">
+        <ul className="resources_list">
+          {resources.map((resource, idx) => (
+            <li key={idx}>
+              <div className="image">
+                <img src={resource.logo} alt={resource.name} />
+              </div>
+              <div className="header">{resource.name}</div>
+              <div>{resource.description}</div>
+              <div className="link">
+                <a href={resource.link} target="_blank" rel="noreferrer">
+                  {resource.name}
+                </a>
+              </div>
+            </li>
+          ))}
```

Comment:
You can stil extract this into a function, if you want. What I meant was inlining the `map` but keeping the `renderResourceFunction()`
```suggestion
          {resources.map(renderResource)}
```

Reply:
can you clarify on what you mean by inlining

Reply:
Inlining is the opposite of extracting. In your original function you extracted everything related to rendering resources (including the `.map()`) into a separate function. I'm just suggesting you keep the map inlined (inside the component code) and extract the function that renders the individual resource.

So here's the change that I want you to do:
```suggestion
          // your function at the top of the file
          function renderResource(resource, idx) => {
            return (
              <li key={idx}>
                <div className="image">
                  <img src={resource.logo} alt={resource.name} />
                </div>
                <div className="header">{resource.name}</div>
                <div>{resource.description}</div>
                <div className="link">
                  <a href={resource.link} target="_blank" rel="noreferrer">
                    {resource.name}
                  </a>
                </div>
              </li>
            )
          }
          
          // the usage in the component
          {resources.map(renderResource)}
```